nimble/host/src/ble_gattc.c

```diff
@@ -912,7 +912,14 @@ ble_gattc_proc_matches_conn_rx_entry(struct ble_gattc_proc *proc, void *arg)
     criteria->matching_rx_entry = ble_gattc_rx_entry_find(
         proc->op, criteria->rx_entries, criteria->num_rx_entries);
 
-    return 1;
+    if (criteria->matching_rx_entry) {
+        /* Return 1 in case we find an entry only
```

Comment:
this comment is not needed.
And actually return (criteria->matching_rx_entry != NULL) would be enough.